oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java

```diff
@@ -613,6 +623,57 @@ private void runWhenPermitted() {
         }
     }
 
+    private boolean shouldProceed() {
+        boolean shouldProceed;
+        NodeState asyncNode = store.getRoot().getChildNode(":async");
+        /*
+            If /:async node already have the lane(under consideration) info, we can proceed ahead, as
+            majorly this change is to stop repository traversal on very first run. If lane had already
+            traversed nodes in repository there is no point stopping this now.
+         */
+        if (asyncNode.exists()) {
+            if (asyncNode.getProperty(name) != null) {
+                shouldProceed = true;
+            } else {
+                shouldProceed = traverseNodesIfLaneNotPresentInIndex || isIndexWithLanePresent();
+            }
+        } else {
+            shouldProceed = traverseNodesIfLaneNotPresentInIndex || isIndexWithLanePresent();
+        }
+        return shouldProceed;
+    }
```

Comment:
While this works, I think it would be simpler to write:

```
NodeState asyncNode = store.getRoot().getChildNode(":async");
if (asyncNode.exists() && asyncNode.hasPropertyName(name)) {
    return true;
}
return traverseNodesIfLaneNotPresentInIndex || isIndexWithLanePresent();
```